### PR TITLE
EDU-165 Add rooms tab

### PR DESCRIPTION
**Pull request by adrianaluchian**
Closes https://educandu.atlassian.net/browse/EDU-165

### CODE REVIEW

File: src/components/pages/import-batch-view.js [hunk 1]
```diff
@@ -155,7 +155,7 @@ function ImportBatchView({ initialState, PageTemplate }) {
         </Row>
         <Row>
           <Space>
-            <span>{t('createdBy')}:</span>
+            <span>{t('common:createdBy')}:</span>
```
Comment (ahelmberger) [inline src/components/pages/import-batch-view.js:158]: :+1:

File: src/components/rooms-tab.js [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+      sorter: by(x => x.access),
+      width: '150px'
+    }
+  ];
+
+  return (
+    <div className="RoomsTab">
+      <section>
+        <h2 className="RoomsTab-sectionHeader">{t('ownedRoomsHeader')}</h2>
+        <Table dataSource={getOwnedRooms(user)} columns={ownedRoomsColumns} size="middle" />
```
Comment (laurentiu-ilici) [inline src/components/rooms-tab.js:143]: I know I've used tables in the project, but we should probably stay away from them when it comes to pages where the normal users operate, they don't really work on mobile. Should we discuss this and what to actually use?

File: src/components/rooms-tab.js [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+      : <span>{room.owner.username}</span>;
+  };
+
+  const renderJoinedOn = (_value, room) => {
+    const userAsMember = room.members.find(member => member.userId === user._id);
+    return <span>{formatDate(userAsMember.joinedOn)}</span>;
+  };
+
+  const renderAccess = (_value, room) => {
+    return <span>{t(`accessType_${room.access}`)}</span>;
```
Comment (laurentiu-ilici) [inline src/components/rooms-tab.js:73]: The _value is already the access level as defined by the data key in the column below

File: src/components/rooms-tab.js [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+      title: t('joinedOn'),
+      dataIndex: 'joinedOn',
+      key: 'joinedOn',
+      render: renderJoinedOn,
+      defaultSortOrder: 'descend',
+      sorter: by(x => x.joinedOn),
+      width: '150px'
+    },
+    {
+      title: t('access'),
```
Comment (ahelmberger) [inline src/components/rooms-tab.js:130]: It's questionable if we wanna show this column as there is no membership in public rooms anyway. But we can keep it for now until we are further into the epic.

File: src/components/rooms-tab.js [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+
+  const renderName = (_value, room) => {
+    return <a href={fakeGetRoomUrl(room._id)}>{room.name}</a>;
+  };
+
+  const renderCreatedOn = (_value, room) => {
+    return <span>{formatDate(room.createdOn)}</span>;
+  };
+
+  const renderOwner = (_value, room) => {
```
Comment (laurentiu-ilici) [inline src/components/rooms-tab.js:61]: The _value is the owner object I believe.

File: src/components/pages/import-batch-view.yml [hunk 6]
```diff
@@ -19,15 +19,9 @@ taskStatusFailed:
 taskStatusDoneWithWarnings:
   en: Done (with warnings)
   de: Abgeschlossen (mit Warnungen)
-createdOn:
-  en: Created on
-  de: Erstellt am
 completedOn:
```
Comment (ahelmberger) [inline src/components/pages/import-batch-view.yml:22]: Maybe this is something that could go into `common` too, what do you think? It's very general.

File: src/components/rooms-tab.js [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+    return room.owner.email
+      ? <span>{room.owner.username} | <a href={`mailto:${room.owner.email}`}>{t('email')}</a></span>
+      : <span>{room.owner.username}</span>;
+  };
+
+  const renderJoinedOn = (_value, room) => {
```
Comment (laurentiu-ilici) [inline src/components/rooms-tab.js:67]: You could make the column's dataInde be "members" and ger skip the second param

File: src/components/rooms-tab.js [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+    members: [{ userId: user._id, joinedOn: new Date().toISOString() }]
+  }
+];
+
+function RoomsTab() {
+  const user = useUser();
+  const { formatDate } = useDateFormat();
+  const { t } = useTranslation('roomsTab');
+
+  const renderName = (_value, room) => {
```
Comment (laurentiu-ilici) [inline src/components/rooms-tab.js:53]: The _value is already the createdOn field

File: src/components/rooms-tab.js (hunk 2 above)
Comment (ahelmberger) [reply src/components/rooms-tab.js:143]: I think we can do a big table overhaul one day. Same goes for the search results. I wouldn't put too much thought into it right now, we don't know what else has to go on these pages yet.

File: src/components/rooms-tab.js (hunk 4 above)
Comment (adrianaluchian) [reply src/components/rooms-tab.js:130]: Yes it's hard to tell for me without real data so far, so I'd also keep this as a helper and reduce it later if needed.

File: src/components/rooms-tab.js (hunk 2 above)
Comment (adrianaluchian) [reply src/components/rooms-tab.js:143]: Agree, I wouldn't deviate from the pattern with this ticket. We already have the Search page which is all users facing and using a table.
We should revisit the layout choices independently so to do it consistently.

File: src/components/pages/import-batch-view.yml (hunk 6 above)
Comment (adrianaluchian) [reply src/components/pages/import-batch-view.yml]: True, it's used on 2 import pages.
Extracted.

**Review by ahelmberger — APPROVED** [review]

File: src/components/rooms-tab.js (hunk 3 above)
Comment (adrianaluchian) [reply src/components/rooms-tab.js]: True! I perpetuated the inconsistencies and confusions to this page 🤦‍♀️ 
Corrected, thanks!

File: src/components/rooms-tab.js (hunk 5 above)
Comment (adrianaluchian) [reply src/components/rooms-tab.js]: Changed

File: src/components/rooms-tab.js (hunk 7 above)
Comment (adrianaluchian) [reply src/components/rooms-tab.js]: Didn't get this comment at first 😬 
I did one better here actually, I mapped columns to rows, as that would be the proper way to do it anyway.

File: src/components/rooms-tab.js (hunk 8 above)
Comment (adrianaluchian) [reply src/components/rooms-tab.js]: Yup, changed.

File: src/components/rooms-tab.js (hunk 3 above)
Comment (adrianaluchian) [reply src/components/rooms-tab.js]: By the way, I streamlined and corrected the other tables too.

File: src/components/rooms-tab.yml [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+  de: Name
+access:
+  en: Access
+  de: Zugang
+joinedOn:
+  en: Joined on
+  de: Eingetreten am
+owner:
+  en: Owner
+  de: Besitzer
```
Comment (adrianaluchian) [inline src/components/rooms-tab.yml:24]: @ahelmberger Would this be the proper translation for room owner?

Comment (laurentiu-ilici) [reply src/components/rooms-tab.yml:24]: I used Eigentümer, if so, I'll also change my PR to have consistency

File: src/components/rooms-tab.js (hunk 3 above)
Comment (ahelmberger) [reply src/components/rooms-tab.js]: :+1:

**Review by laurentiu-ilici — APPROVED** [review]

File: src/components/rooms-tab.yml (hunk 9 above)
Comment (adrianaluchian) [reply src/components/rooms-tab.yml]: Extracted to `owner` to `common`

**Pull request merged**